### ISSUE TITLE
Should the STUDENTS variable be an @@all?

### ISSUE THREAD

**Issue by rossmawd** [learn-co-curriculum/sinatra-nested-forms-readme#27]
Won't Ruby try to warn you every time you create a new instance of Student? As Ruby will think that STUDENTS is a constant?

**Comment by Sdcrouse** [comment]
I think I see what you're getting at. Right before #initialize, you have this line of code: `STUDENTS = []`. I tried out that Student class in IRB. Ruby didn't warn me about anything, except for when I directly changed the STUDENTS variable like this: `Student::STUDENTS = []`, and when I set it a second time inside of the Student class. Then, I got the warning that you expected.

I think what happens is that, like class variables, constants are only ever set _**one**_ time: when the class itself is first created; Ruby will _**not**_ reset the constants or class variables every time a new object is created. I haven't yet found a resource that talks about it, but I'm 99% certain that this is what's going on.

If you're simply adding a new student to the STUDENTS array (as is the case with the #initialize method), then Ruby won't object to that. Ruby only seems to send a warning when you reset the constant from outside the class, or from inside the class (but _**outside**_ of any methods - you get a SyntaxError for dynamic constant assignment if you try to change STUDENTS from inside of a class method or instance method).

I hope that helps!

---Sdcrouse
